flake.nix

```diff
@@ -20,12 +20,14 @@
         "armv7l-linux"
         "aarch64-darwin"
       ];
-
-      forAllSystems = f: lib.genAttrs systems (system: f system);
-
     in
     {
       lib = lib.extend (final: prev: {
+        inherit (import ./lib/flake-utils.nix {
+          nixpkgs = self;
+          inherit systems;
+        }) eachSystem eachDefaultSystem;
```

Comment:
I would suggest `for` instead of `each`. It's the closest thing we have to a standard name for `flip map`.

It matches Haskell's `for = flip traverse` except no monad. It's also close to C-style languages' `for` keyword which is appropriate here.
I can't think of similar connections for `each`.

Reply:
Do you mean `forSystem`? I am not a haskell person like the most of us. I would understand `mapSystems` instead of `eachSystem` but `forSystem` does not sound intuitive.

Reply:
javscript developer to the giants:
`for (const item of array)` is very common pattern in javscript, so there would be a big marketshare for intuitive understanding of the prefix "for" in the context of list and iterator.

Reply:
if we disagree, we could use both? There's a precedence for that https://ramdajs.com/docs/#forEach

Reply:
Well, each and forEach both represent this function. But they are both impure side-effect based functions. mapSystems makes therefore more sense for this.

Reply:
`mapSystems` it is.